### PR TITLE
Friendship milestone v2

### PR DESCRIPTION
**Pull request by jstloyal**
- Add two-way unique  friendship
working

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -12,27 +12,26 @@ class User < ApplicationRecord
...
+  has_many :friend_requests, through: :inverted_friendships, source: :user
 
   def confirm_friend(user)
-    friendship = inverse_friendships.find { |friend| friend.user == user }
+    friendship = inverse_friendships.where(user_id: user.id).first
     friendship.status = true
     friendship.save
+    reverse_confirmation = friendships.build(friend_id: user.id, status: true)
+    reverse_confirmation.save
   end
```
Comment (stratospherique) [inline app/models/user.rb:28]: * [ ] You have the same method (better) in Friendship model, kindly delete this one.

File: app/models/user.rb [hunk 2]
```diff
@@ -12,27 +12,26 @@ class User < ApplicationRecord
...
 
   def friend?(user)
     friends.include?(user)
   end
```
Comment (stratospherique) [inline app/models/user.rb:32]: * [ ] This method is unnecessary, the friendship was handled by associations.

File: app/models/user.rb (hunk 1 above)
Comment (jstloyal) [reply app/models/user.rb:28]: Please where in the friendship model are you referring to (confirmed_friend)?

**Comment by jstloyal** [comment]
show-page

**Comment by Anna-Myzukina** [comment]
# [APPROVED]  :heavy_check_mark: 

## Thanks for applying the changes  :+1:

### Go and merge to development.

# :raising_hand:  See you at the next projects.

Happy coding  <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Pull request merged**